idseq_dag/engine/pipeline_step.py

```diff
@@ -113,6 +116,54 @@ def wait_for_input_files(self):
                         time.sleep(5)
             self.input_files_local.append(flist)
 
+    def validate_input_files(self):
+        ''' Validate input files before running the step '''
+        errors = self.get_input_file_validation_errors()
+        if errors and "error_type" in errors and errors["error_type"] != InputErrorType.NO_ERROR:
+            log.write("Invalid input detected for step %s" % self.name)
+            self.write_input_errors_json({
+                "step": self.name,
+                "errors": errors["errors"] if "errors" in errors else [],
+                "error_type": errors["error_type"].name
+            })
+            raise RuntimeError("input files for step %s were not valid" % self.name)
+
+    def get_input_file_validation_errors(self):
+        ''' Return any input file validation errors'''
+        # Steps should overwrite this method if necessary.
+        return {
+            # Step-specific error messages.
+            "errors": [],
+            # The broad type of error that occurred.
+            "error_type": InputErrorType.NO_ERROR
```

Comment:
I don't think you need a `NO_ERROR` state. If you follow my suggestion above, just decide whether to put a file in `additional_files_to_upload` or not based on whether there was actually an error or not

Reply:
I can get rid of `NO_ERROR`. The reason for creating an `InputErrorType` enum in the first place was I assumed we would eventually have a couple of different error types (another one is MALFORMED_INPUT_FILES and we might divide that one into subcategories).